server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/SendIssueNotificationsStep.java

```diff
@@ -185,9 +187,17 @@ public String getDescription() {
     return "Send issue notifications";
   }
 
+  @CheckForNull
   private String getBranchName() {
     Branch branch = analysisMetadataHolder.getBranch();
-    return branch.isMain() ? null : branch.getName();
+    return branch.isMain() || branch.getType() == PULL_REQUEST ? null : branch.getName();
+  }
+
+  @CheckForNull
+  private String getPullRequest() {
+    Branch branch = analysisMetadataHolder.getBranch();
+    // TODO rely on the pull request instead of the branch name
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fbranch%2Ffirst-class-pull-request&issues=AWGuRf0vaHWQ5zPschp1&open=AWGuRf0vaHWQ5zPschp1)

[ISSUE_KEY]: AWGuRf0vaHWQ5zPschp1